pkg/trace/agent/agent_test.go

```diff
@@ -581,3 +587,147 @@ func formatTrace(t pb.Trace) pb.Trace {
 	}
 	return t
 }
+
+func BenchmarkThroughput(b *testing.B) {
+	if _, ok := os.LookupEnv("DD_TRACE_TEST_PRIVATE_DATA"); !ok {
+		b.SkipNow()
+	}
+	b.Run("2.msgp", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/2.msgp"))
+	b.Run("3.msgp", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/3.msgp"))
+	b.Run("4.msgp", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/4.msgp"))
+	b.Run("5.msgp", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/5.msgp"))
+	b.Run("6.msgp", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/6.msgp"))
+}
+
+func benchThroughput(file string) func(*testing.B) {
+	return func(b *testing.B) {
+		in, err := os.Open(filepath.Join(os.Getenv("GOPATH"), file))
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer in.Close()
+		// prepare the traces in this file by adding sampling.priority=2
+		// everywhere to ensure consistent sampling assumptions and results.
+		var traces pb.Traces
+		if err := msgp.Decode(in, &traces); err != nil {
+			b.Fatal(err)
+		}
+		var count int
+		for _, t := range traces {
```

Comment:
Shouldn't we also rewrite the timestamps of the traces? Otherwise if I'm not mistaken the agent simply drops all traces older than X.

Reply:
Benjamin changed that in some PR a while ago. Now we can accept no matter how far in the past. Also notice that in each benchmark iteration we wait on the "writer channel" until _all_ the traces are sent out, which confirms that they all go through full processing.

Reply:
Ref. https://github.com/DataDog/datadog-trace-agent/pull/493

Reply:
Oh nice, I wasn't aware of that change. It sounds good then 👍